src/auro-themeswitcher.js

```diff
@@ -0,0 +1,236 @@
+
+    this.currentTheme = this.newTheme;
+
+    this.currentTheme.forEach((theme) => {
+      const themeObj = JSON.parse(theme);
+      const link = document.createElement("link");
+
+      link.setAttribute('theme', themeObj.label);
+      link.rel = "stylesheet";
+      link.href = themeObj.url;
+
+      head.appendChild(link);
+    })
+  }
+
+  handleCheckboxSelection(evt) {
+    if (evt.target.checked) {
+      const themeIndex = this.newTheme.indexOf(evt.target.value);
+
+      if (themeIndex === -1) {
+        this.newTheme.push(evt.target.value);
+      }
+    } else {
+      const themeIndex = this.newTheme.indexOf(evt.target.value);
+
+      if (themeIndex > -1) {
+        this.newTheme.splice(themeIndex, 1);
+      }
+    }
+
+    this.requestUpdate();
+    this.handleApplyBtnState();
+  }
+
+  handleApplyBtnState() {
+    this.getLoadedThemes();
+
+    if (JSON.stringify(this.loadedThemes) === JSON.stringify(this.newTheme)) {
+      this.disableApply = true;
+    } else {
+      this.disableApply = false;
+    }
+  }
+
+  applyThemes() {
+    this.disableApply = true;
+
+    this.toggleThemeSwitcher();
+    this.unloadThemes();
+    this.loadSelectedthemes();
+  }
+
+  markLoadedthemes() {
+    this.loadedThemes.forEach((theme) => {
+      const checkboxes = this.shadowRoot.querySelector('auro-dialog').querySelectorAll('auro-checkbox');
+
+      checkboxes.forEach((checkbox) => {
+        if (checkbox.id === theme.label) {
+          checkbox.setAttribute('checked', true);
+
+          this.newTheme.push(checkbox.getAttribute('value'));
+          this.currentTheme.push(checkbox.getAttribute('value'));
+        }
+
+        this.requestUpdate();
+      })
+    })
+  }
+
+  toggleThemeSwitcher() {
+    const dialog = this.shadowRoot.querySelector('#auroThemeSwitcherDialog');
+
+    dialog.hasAttribute('open') // eslint-disable-line no-unused-expressions
+    ? dialog.removeAttribute("open")
+    : (dialog.removeAttribute("open"),
+      dialog.setAttribute("open", true))
+  }
+
+  firstUpdated() {
+    this.dropdown = this.shadowRoot.querySelector('#auroThemeSelector');
+
+    this.getLoadedThemes();
+    this.markLoadedthemes();
+  }
+
+  render() {
+    return html`
+      <div>
+        <auro-button class="themeSwitcherLaunchBtn" aria-label="Theme Switcher" variant="tertiary" slim @click="${this.toggleThemeSwitcher}">
+          <auro-icon category="interface" name="settings"></auro-icon>
+        </auro-button>
+        <auro-dialog id="auroThemeSwitcherDialog">
+          <span slot="header">Theme Switcher</span>
+          <div slot="content">
+            <div class="selector">
+              <span slot="legend">
+                Choose which themes to apply:
+              </span>
+              <auro-checkbox-group required>
+                ${this.themes.map((theme) => html`
+                  <auro-checkbox
+                    value="${JSON.stringify(theme)}"
```

Comment:
The only 2 themes are Auro and Jetstream. Darkmodes are modes contained within the main theme. 

See what [spectrum](https://opensource.adobe.com/spectrum-web-components/) did. 
<img width="326" alt="Screen Shot 2023-09-25 at 2 31 46 PM" src="https://github.com/AlaskaAirlines/auro-button/assets/181089/c452be6d-e144-411b-b9a6-d78cac2bcd41">